<!DOCTYPE html>
<html>

<head>
	<title>CRISPR editing check</title>
	<style>
		body{
			margin: 0 auto;
			font-size: 12pt;    
			/*font-family: 'PT Serif', serif;*/
			line-height: 1.2;
			max-width: 8in;
			/*text-align: justify;*/
		}
	</style>
</head>

<body>
	<h1>CRISPR Editing Analysis</h1>

<p style="color:red;">!!! Make sure both the left and the right flanking sequences are in the same read or merged read in the fastq file !!!</p>

	<label for="left">Your left flanking sequence:</label>
	<input id="left" name="LeftSequence" placeholder="a short sequence left of the gRNA" size="40"><br><br>
	<label for="right">Your right flanking sequence:</label>
	<input id="right" name="RightSequence" placeholder="a short sequence right of the gRNA" size="40"><br><br>
	<label for="grna">Your guide RNA sequence:</label>
	<input id="grna" name="GuideRNA" placeholder="the gRNA sequence" size="40"><br><br>
	<label for="wt">Wild type sequence</label><br>
	<textarea id="wt" name="Wildtype" rows="5" cols="80" placeholder="The target gene sequences (long enough to include both the left and right flanking sequences)"></textarea>
	<br><br>
	<button onclick="clearseq()">Clear</button>
	<button onclick="putExample()">Example Input</button><br><br>
	<input type="file" id="files" name="files[]" multiple /><br><br>
	<button onclick="analyze()">Start Analyze</button>
	<select id="box1">
		<option value="F">Forward Strand</option>
		<option value="R">Reverse Strand</option>
	</select>
	<label for="box1">(Read1 or Read2? Forward strand is to check read1; reverse strand is for read2)</label><br>
	<progress style="visibility:hidden" id="progress" value="0" max="100"></progress><br>
	<button style="visibility:hidden" id="download-btn" onclick="download()"> Download output csv file</button><br>
	<p id="demo"></p><br>
	<output id="output" style="display:none"></output>

	<h2>Help</h2>
	<p>This program summarize CRISPR editing results based on the idea of <a href='https://github.com/patrickc01/CRIS.py'>CRIS.py</a>. You can read the paper <a href='https://www.nature.com/articles/s41598-019-40896-w'>here</a>. The basic idea is summarized in its Fig. 1. 
	<img src="images/CRISjs.png" alt="CRISjs dialog" width=100%></p>
	<p>All three sequences (the left and right flanking sequences and the gRNA sequences) should be on the <strong>same strand as the template</strong>. If you are working on polyploid species, your left or right flanking sequences should be unique to your template (subgenome) if the fastq files include homolog/homeologs.</p>
	<h3>Step</h3>
	<ol>
	<li>Input all the required sequences. Make sure both the left and the right flanking sequences are in the same read or merged read in the fastq file.</li>
	<li>Select fastq or fastq.gz files by clicking &quot;<strong>Choose files</strong>&quot;.</li>
	<li>Start analyzing by clicking the &quot;<strong>Start Analyze</strong>&quot; button. </li>
	<li>Download the summary file by clicking the &quot;<strong>Download output csv file</strong>&quot; button.</li>
	</ol>
	<p>You can click the &quot;<strong>Example input</strong>&quot; button to get the example inputs (then select the example fastq.gz files <a href='https://github.com/pinbo/CRISjs/tree/main/example-input' target='_blank' class='url'>https://github.com/pinbo/CRISjs/tree/main/example-input</a>).</p>
	<p>If you have a lot of fastq files, you may use the standalone command line program <a href='https://github.com/pinbo/CRISgo'>CRISgo</a>.</p>



	<script src="libs/pako_inflate.min.js"></script>
	<script src="libs/FileSaver.min.js"></script>
	<script src="libs/crisjs-functions.js"></script>

</body>

</html>